libr/debug/p/native/linux/linux_debug.h

```diff
@@ -63,3 +63,165 @@ int linux_reg_write (RDebug *dbg, int type, const ut8 *buf, int size);
 RList *linux_desc_list (int pid);
 int linux_handle_signals (RDebug *dbg);
 const char *linux_reg_profile (RDebug *dbg);
+/* coredump api */
+
+//#include "elf.h"
+#include "elf_specs.h"
+#include <sys/procfs.h>
+
+#define SIZE_PR_FNAME	16
+
+#define ELF_HDR_SIZE sizeof(Elf64_Ehdr)
+
+#define R_DEBUG_REG_T struct user_regs_struct
+
+#define SIZE_NT_FILE_DESCSZ sizeof(unsigned long) * 3   /* start_address * end_address * offset_address */
+                                                        /* 
+                                                        NT_FILE layout:
+                                                        [number of mappings]
+                                                        [page size]
+                                                        [foreach(mapping)
+                                                                [start_address]
+                                                                [end_address]
+                                                                [offset_address]
+                                                        [filenames]
+                                                        */
+#define DEFAULT_NOTE    6
+
+#define X_MEM 0x1
+#define W_MEM 0x2
+#define R_MEM 0x4
+#define P_MEM 0x8
+#define S_MEM 0x10
+#define WRG_PERM 0x20
+
+#define MAP_ANON_PRIV	0x1
+#define MAP_ANON_SHR	0x2
+#define MAP_FILE_PRIV	0x4
+#define MAP_FILE_SHR	0x8
+#define MAP_ELF_HDR	0x10
+#define MAP_HUG_PRIV	0x20
+#define MAP_HUG_SHR	0x40
+
+#define SH_FLAG         0x1
+#define IO_FLAG         0x2
+#define DD_FLAG         0x4
+#define HT_FLAG         0x8
+#define PV_FLAG         0x10 /* just for us */
+
+static unsigned int n_notes = DEFAULT_NOTE;
+
+typedef struct proc_stat_content {
+        int pid;
+        int ppid;
+        int pgrp;
+        int sid;
+        char s_name;
+        ut32 flag;
+	unsigned long int utime;
+	unsigned long int stime;
+	long int cutime;
+	long int cstime;
+        unsigned long int nice;
+        unsigned long int num_threads;
+        unsigned long int sigpend;
+        unsigned long int sighold;
+        ut32 uid;
+        ut32 gid;
+	unsigned char coredump_filter;
+}proc_stat_content_t;
+
+typedef struct map_file {
+        unsigned int count;
+        unsigned int size;
+}map_file_t;
+
+typedef struct auxv_buff {
+        void *data;
+        size_t size;
+} auxv_buff_t;
+
+typedef struct linux_map_entry {
+        ut64 start_addr;
+        ut64 end_addr;
+        ut64 offset;
+        ut64 inode;
+        ut8 perms;
+        bool anonymous;
+	bool dumpeable;
+	bool kernel_mapping;
+	bool file_backed;
+        char *name;
+        struct linux_map_entry *n;
+}linux_map_entry_t;
+
+
+#define ADD_MAP_NODE(p)         do {							\
```

Comment:
try to oneliner macros or make them static inline functions. no need to use do {}while(0). just using {} is enough